cmake/modules/pre_dt.cmake

```diff
@@ -28,6 +28,12 @@ include(extensions)
 # - ZEPHYR_BASE: path to zephyr repository (added to DTS_ROOT)
 # - SHIELD_DIRS: paths to shield definitions (added to DTS_ROOT)
 
+# Include board specific device-tree flags before parsing.
+foreach(dir ${BOARD_DIRECTORIES})
+  include(${dir}/pre_dt_board.cmake OPTIONAL)
```

Comment:
this will cause board directories out of tree to be included even when upstream board is used.
For example `plank/soc` being defined in Zephyr and `plank/soc/oot` defined oot will cause `pre_dt_board.cmake` from both locations to be sourced, regardless of which board variant is used.

This again means that building the same sample for `plank/soc` may behave differently when the oot Zephyr board module is availble or not.

This is not acceptable, as it makes it very hard to predict the behavior of a given board for a sample.